File: src/main/java/org/dedda/bratwurst/parse/ObjectCreationParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.ObjectCreation;

/**
 * Created by dedda on 10/17/15.
 *
 * @author dedda
 */
public class ObjectCreationParser extends ExpressionParser {

    public ObjectCreation parse(String line, int lineNumber) {
        line = line.trim();
        return new ObjectCreation(lineNumber, line.substring(
                Patterns.CLASS_INSTANTIATION_OPEN.length(),
                line.length() - Patterns.CLASS_INSTANTIATION_CLOSE.length()
        ));
    }

}

File: src/main/java/org/dedda/bratwurst/parse/ObjectCreationParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.ObjectCreation

/**
 * Created by dedda on 10/17/15.
 *
 * @author dedda
 */
class ObjectCreationParser : ExpressionParser() {
    override fun parse(line: String, lineNumber: Int): ObjectCreation {
        val trimmed = line.trim()
        return ObjectCreation(lineNumber, trimmed.substring(1, trimmed.length - 1))
    }
}